Automattic/wp-calypso
# ExtensiveLodashReplacementPlugin : Use `nmf.getResolver` to find modules.

Switch to webpack's `enhanced-resolve` through `nmf.getResolver` to find modules.
This saves us from running the algorithm ourselves and parsing JSON.

@jsnajdr suggested this change in #34138. 

Note that the version matching algorithm has been changed, since we no longer have access to the requested version range. It now checks whether the major versions match and the root `lodash-es` is the same or newer than the resolved `lodash` or `lodash.foo`.

#### Changes proposed in this Pull Request

* Use `enhanced-resolve` for module resolution in `ExtensiveLodashReplacementPlugin`
* Rework version comparison logic to allow for comparing direct versions
* Add build-time warning when a package that can't be replaced is detected

#### Testing instructions

* Ensure that the build does not break
* Ensure that chunk sizes remain the same. This is an indicator that all usages of `lodash` and `lodash.foo` are still being correctly replaced.
